test/framework/src/test/java/org/elasticsearch/test/VersionUtilsTests.java

```diff
@@ -0,0 +1,167 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.elasticsearch.test;
+
+import org.elasticsearch.Version;
+import org.elasticsearch.common.collect.Tuple;
+
+import java.util.Arrays;
+import java.util.List;
+
+import static java.util.Collections.emptyList;
+import static java.util.Collections.singletonList;
+
+public class VersionUtilsTests extends ESTestCase {
+
+    public void testAllVersionsSorted() {
+        List<Version> allVersions = VersionUtils.allReleasedVersions();
+        for (int i = 0, j = 1; j < allVersions.size(); ++i, ++j) {
+            assertTrue(allVersions.get(i).before(allVersions.get(j)));
+        }
+    }
+
+    public void testRandomVersionBetween() {
```

Comment:
I just moved this test and the one above it. They were in the `org.elasticsearch.test.test` package which would have forced me to make `VersionUtils. resolveReleasedVersions` public and that seemed wrong.